onnx/defs/math/defs.cc

```diff
@@ -3474,4 +3474,205 @@ ONNX_OPERATOR_SET_SCHEMA(
           result_shape_proto.add_dim()->set_dim_value(2);
           updateOutputShape(ctx, 0, result_shape_proto);
         }));
+
+static const char* SVD_doc = R"DOC(
+Singular Value Decomposition
+
+Factorizes A using SVD. Both full and partial SVD compute the standard SVD factorization `A = U * diag(S) * Vh`.
+Computes the full SVD by default. Partial SVD is also known as [thin SVD](https://en.wikipedia.org/wiki/Singular_value_decomposition#Thin_SVD)
+
+Setting compute_uv to false (0) will only compute the singular values. For the singular values, full vs partial SVD does not apply as the singular values 
+will always have the same dimensions.
+
+A must have at least 2 dimensions.
+
+All dimensions higher than 2 are used for broadcasting, and SVD is only applied to the lowest two dimensions.
+See [numpy](https://numpy.org/doc/stable/reference/generated/numpy.linalg.svd.html).
+
+Dimensions:
+
+A: (M, N)
+
+K: Min(M, N)
+
+Full SVD:
+
+U: (M, M)
+
+S: (K)
+
+Vh: (N, N)
+
+Partial SVD:
+
+U: (M, K)
+
+S: (K)
+
+Vh: (K, N)
+
+Numerical stability of derivative:
+There is no numerical instability in the derivative when only computing the singular values.
+
+The derivative of the full factorization is dependent on the reciprocal of each singular value and the
+reciprocal of the differences of the squares each pair of singular values.
+
+This causes numerical instability when,
+1. Any singular value is 0.
+2. Any two singular values are the same.
+3. Any two singular values are sufficiently close.
+)DOC";
+
+void SVDInferenceFunction(InferenceContext& ctx) {
```

Comment:
I wonder if this function works if one dimension is null or specified with a letter.

Reply:
Thank you for the catch! I missed this was a possibility in the docs. 

I added checks for when the shape is not present and for computing the number of singular values when one or both of the non-batched input dimensions are not present and/or symbolic. 

I added tests here https://github.com/onnx/onnx/pull/4416/commits/bbed9b2389ea8f589aa579969c0bcc45399f80bd#diff-1109925d18b450cb60dcd1438f23475daad61de5eca71acdcc112b3aa85a7c0aR8612-R8794